trunk/src/app/srs_app_recv_thread.cpp

```diff
@@ -577,12 +577,12 @@ srs_error_t SrsHttpRecvThread::cycle()
     srs_error_t err = srs_success;
     
     while ((err = trd->pull()) == srs_success) {
-        ISrsHttpMessage* req = NULL;
-        SrsAutoFree(ISrsHttpMessage, req);
-        
-        if ((err = conn->pop_message(&req)) != srs_success) {
+        ISrsHttpMessage* req_raw = NULL;
+        if ((err = conn->pop_message(&req_raw)) != srs_success) {
```

Comment:
https://github.com/ossrs/srs/blob/baf22d01c1a0f22ae5739f94664763d7b92b173f/trunk/src/app/srs_app_http_conn.cpp#L338

This method will always sent NULL message out.
Suggest refactor `pop_message` or use `NULL to replace `req_raw`.

```suggestion
        if ((err = conn->pop_message(NULL)) != srs_success) {
```

Reply:
Fixed.